packages/admin-next/dashboard/src/routes/products/product-create/components/product-create-details-form/components/product-create-details-media-section/product-create-details-media-section.tsx

```diff
@@ -1,69 +1,82 @@
+import { UploadMediaFormItem } from "../../../../../common/upload-media-form-item"
+import { MediaGrid } from "../../../../../common/media-grid-view"
+import { useCallback, useState } from "react"
 
 type ProductCreateMediaSectionProps = {
   form: UseFormReturn<ProductCreateSchemaType>
 }
 
-const SUPPORTED_FORMATS = [
-  "image/jpeg",
-  "image/png",
-  "image/gif",
-  "image/webp",
-  "image/heic",
-  "image/svg+xml",
-]
-
 export const ProductCreateMediaSection = ({
   form,
 }: ProductCreateMediaSectionProps) => {
   const { t } = useTranslation()
+  const [selection, setSelection] = useState<Record<string, true>>({})
+  const selectionCount = Object.keys(selection).length
+
+  const { fields, append, remove } = useFieldArray({
+    name: "media",
+    control: form.control,
+    keyName: "field_id",
+  })
+
+  const handleDelete = () => {
+    const ids = Object.keys(selection)
+    const indices = ids.map((id) => fields.findIndex((m) => m.id === id))
+
+    remove(indices)
+    setSelection({})
+  }
+
+  const handleCheckedChange = useCallback(
+    (id: string) => {
+      return (val: boolean) => {
+        if (!val) {
+          const { [id]: _, ...rest } = selection
+          setSelection(rest)
+        } else {
+          setSelection((prev) => ({ ...prev, [id]: true }))
+        }
+      }
+    },
+    [selection]
+  )
 
   return (
     <div id="media" className="flex flex-col gap-y-8">
       <Heading level="h2">{t("products.media.label")}</Heading>
       <div className="grid grid-cols-1 gap-x-4 gap-y-8">
-        <Form.Field
-          control={form.control}
-          name="images"
-          render={() => {
-            return (
-              <Form.Item>
-                <div className="flex flex-col gap-y-4">
-                  <div className="flex flex-col gap-y-1">
-                    <Form.Label optional>
-                      {t("products.media.label")}
-                    </Form.Label>
-                    <Form.Hint>{t("products.media.editHint")}</Form.Hint>
-                  </div>
-                  <Form.Control>
-                    <FileUpload
-                      label={t("products.media.uploadImagesLabel")}
-                      hint={t("products.media.uploadImagesHint")}
-                      hasError={!!form.formState.errors.images}
-                      formats={SUPPORTED_FORMATS}
-                      onUploaded={(files) => {
-                        form.clearErrors("images")
-                        // if (hasInvalidFiles(files)) {
-                        //   return
-                        // }
-
-                        // files.forEach((f) => append(f))
-                      }}
-                    />
-                  </Form.Control>
-                  <Form.ErrorMessage />
-                </div>
-              </Form.Item>
-            )
-          }}
-        />
+        <UploadMediaFormItem form={form} append={append} />
       </div>
+      {fields?.length ? (
```

Comment:
The UI can be revised, but I kept it consistent with how images are handled in the edit flow

Reply:
I think we should ping Ludvig on this one, this doesn't look to great but fine for now, and then we can follow up in another PR.

One todo is that there is currently no way to set an image as the thumbnail.